File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
       int min_val = prices[0];
       int temp = 0;
       int r= 0;
       int ans = 0;
       while(r<prices.size())
       {
            
            temp = prices[r]-min_val;
            ans = max(ans,temp); 
            if(prices[r]<min_val){
                ans = max(ans,temp);
                temp = 0;
                min_val = prices[r];
                
            }
            r++;
       }
       return ans;
    }
};

File: Data Structures & Algorithms/climbing-stairs/submission-0.py
class Solution:
    def climbStairs(self, n: int) -> int:

        def f(i):
            if i <= 1:
                return 1
            l = f(i-1)
            r = f(i-2)
            return l+r
        ans = f(n)
        return ans
                

        